### ISSUE TITLE
Retention rule gone?

### ISSUE THREAD

**Issue by mapagm** [nextcloud/files_retention#118]
Hello,

I’ve created an invisible tag and automatic tagging rule. Then (I think) I’ve created retention rule: selected previously created tag and set to delete files after 30 days and when i clicked create button nothing happened… How to find out if creted rule exist and is active? Now when I try to create this rule again, there is no such tag in search box. I have nextcloud 20.0.9 and Retention 1.9.0.

Greetings,
Martin

**Comment by japerrin** [comment]
Hi,

I have exactly the same issue with nextcloud 21.0.0 and Retention 1.10.0.

Any help will be helpful

Thanks
Jacques

**Comment by japerrin** [comment]
I found the fix by myself by removing entries linked to a deleted tag in the table oc_retention.

1) Open a SSH session on Nextcloud server with use ncadmin
2) Run psql (my database is Postgres)

psql -dnextcloud_db
psql (12.6 (Ubuntu 12.6-0ubuntu0.20.04.1))
Type "help" for help.

3) List tags used in retention rules

SELECT * FROM oc_systemtag;
 id |        name         | visibility | editable
----+---------------------+------------+----------
  3 | filetype:htmlreport |          1 |        1
  6 | filetype:vropackage |          1 |        1
(2 rows)

4) List entries in the retention table
 SELECT * FROM oc_retention;
 id | tag_id | time_unit | time_amount
----+--------+-----------+-------------
  3 |      3 |         3 |           2
  5 |      5 |         3 |           1
  6 |      6 |         3 |           1


5) Then delete rules linked to non existing tags:
 DELETE FROM oc_retention WHERE id = 5;


Refresh in Nextcloud settings "et voila" !!!

**Comment by nickvergessen** [comment]
Should be fixed automatically by #135

**Comment by nickvergessen** [comment]
Problem is a JS error:
```
ncaught TypeError: _this.tagCollection.get(...) is undefined
    render retentionview.js:61
    He Underscore
    i Backbone
    render retentionview.js:58
    init admin.js:61
    jQuery 11
    Backbone 4
    init admin.js:53
    <anonymous> admin.js:168
    jQuery 12
    Webpack 7
```
